optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/TaskRepositoryES.java

```diff
@@ -130,6 +135,25 @@ public boolean tryDeleteByQueryRequest(
                     .waitForCompletion(false)
                     .conflicts(Conflicts.Proceed));
 
+    if (configuration.getConnection().isHealthCheckEnabled()) {
+      return async(query, deletedItemIdentifier, request);
+    } else {
+      return sync(request);
+    }
+  }
+
+  private boolean sync(final DeleteByQueryRequest request) {
+    try {
+      final Long deleted = esClient.submitDeleteTask(request).deleted();
```

Comment:
I don't think this is a sync request because the property `waitForCompletion` is still marked as `false`. Would we not also need to set the flag there depending on whether we go for sync or async requests?

Reply:
fixed